### ISSUE TITLE
New `unusedArguments` bug in 0.55

### ISSUE THREAD

**Issue by calda** [nicklockwood/SwiftFormat#1942]
When updating our SwiftFormat build to 0.55.3, I found two cases where the `unusedArguments` rule now unexpectedly removes an argument that is actually used. Here are two test cases that currently fail on `develop`:

```swift
func testShadowedArgumentNameInGuard() {
    let input = """
    func getTableInfo(forTable table: String) {
        guard let table = try? db.schema.objectDefinitions(name: table, type: .table).first else { return nil }
        print(table)
    }
    """

    testFormatting(for: input, rule: .unusedArguments)
}

func testShadowedArgumentNameInDoBlock() {
    let input = """
    func mapToResponse(_ jsonData: Data) -> Response {
        let jsonData = try! JSONSerialization.jsonObject(with: jsonData, options: []) as! [String: AnyObject]
        return Response(jsonData)
    }
    """

    testFormatting(for: input, rule: .unusedArguments)
}
```

**Comment by nicklockwood** [comment]
Thanks for the test cases - super helpful! ❤️

**Comment by nicklockwood** [comment]
@calda fixed in  0.55.4
